### PR TITLE
client: do not use g_conf().get_val<>() in libcephfs

### PR DESCRIPTION
**Pull request by lxbsz**
The g_conf() will try to dereference the "g_ceph_context" to get
the "_conf", but the "g_ceph_context" won't be set in none fuse
use case in libcephfs, so it will crash in when doing
g_conf().get_val<>.

Switch to use cct->_conf.get_val<> instead.

Fixes: https://tracker.ceph.com/issues/48206
Signed-off-by: Xiubo Li <xiubli@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by tchaikov — APPROVED** [review]

**Review by batrick — CHANGES_REQUESTED** [review]
Please also fix `get_fuse_groups` in `fuse_ll.cc`. (It won't crash there but we should be consistent.)

**Comment by lxbsz** [comment]
> Please also fix `get_fuse_groups` in `fuse_ll.cc`. (It won't crash there but we should be consistent.)

Sure, done. Thanks.

**Comment by lxbsz** [comment]
jenkins test api

**Review by batrick — APPROVED** [review]

**Comment by batrick** [comment]
jenkins test api

File: src/client/fuse_ll.cc [hunk 1]
```diff
@@ -136,7 +136,7 @@ static int getgroups(fuse_req_t req, gid_t **sgids)
 
 static void get_fuse_groups(UserPerm& perms, fuse_req_t req)
 {
-  if (g_conf().get_val<bool>("fuse_set_user_groups")) {
+  if (cct->_conf.get_val<bool>("fuse_set_user_groups")) {
```
Comment (batrick) [inline src/client/fuse_ll.cc:139]:
```suggestion
  if (cfuse->client->cct->_conf.get_val<bool>("fuse_set_user_groups")) {
```

Fixes build error:

````
/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/16.0.0-7334-gdb29fdf8/rpm/el8/BUILD/ceph-16.0.0-7334-gdb29fdf8/src/client/fuse_ll.cc: In function 'void get_fuse_groups(UserPerm&, fuse_req_t)':
/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/16.0.0-7334-gdb29fdf8/rpm/el8/BUILD/ceph-16.0.0-7334-gdb29fdf8/src/client/fuse_ll.cc:139:7: error: 'cct' was not declared in this scope
   if (cct->_conf.get_val<bool>("fuse_set_user_groups")) {
       ^~~
/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/16.0.0-7334-gdb29fdf8/rpm/el8/BUILD/ceph-16.0.0-7334-gdb29fdf8/src/client/fuse_ll.cc:139:7: note: suggested alternative: 'cc_t'
   if (cct->_conf.get_val<bool>("fuse_set_user_groups")) {
       ^~~
       cc_t
/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/16.0.0-7334-gdb29fdf8/rpm/el8/BUILD/ceph-16.0.0-7334-gdb29fdf8/src/client/fuse_ll.cc:139:26: error: expected primary-expression before 'bool'
   if (cct->_conf.get_val<bool>("fuse_set_user_groups")) {
                          ^~~~
/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/16.0.0-7334-gdb29fdf8/rpm/el8/BUILD/ceph-16.0.0-7334-gdb29fdf8/src/client/fuse_ll.cc:139:26: error: expected ')' before 'bool'
   if (cct->_conf.get_val<bool>("fuse_set_user_groups")) {
      ~                   ^~~~
                          )
````

**Review by batrick — CHANGES_REQUESTED** [review]

File: src/client/fuse_ll.cc (hunk 1 above)
Comment (lxbsz) [reply src/client/fuse_ll.cc]: Done. Thanks.

**Comment by batrick** [comment]
jenkins test docs

**Comment by batrick** [comment]
jenkins test api

**Review by batrick — APPROVED** [review]

**Comment by batrick** [comment]
https://pulpito.ceph.com/pdonnell-2020-12-02_07:09:18-fs-wip-pdonnell-testing-20201202.050726-distro-basic-smithi/

**Pull request merged**